sky/serve/controller.py

```diff
@@ -158,7 +158,7 @@ def configure_logger():
         logger.info('SkyServe Controller started on '
                     f'http://localhost:{self._port}')
 
-        uvicorn.run(self._app, host='localhost', port=self._port)
+        uvicorn.run(self._app, host='0.0.0.0', port=self._port)
```

Comment:
Hmm, this is required for HA for external load balancers to query the controller for replica ips.

How about we:
1. Keep `host=0.0.0.0`, but do not expose port 20001 through SkyPilot. Thus by default, the controller port is not accessible.
2. For HA, we create an additional k8s internal ClusterIP service to port `20001-20020` on the controller in our `create_or_update_lb.sh` script. This will not be exposed to public, but can be used by HA load balancers to communicate with the controller.